### PR TITLE
Add SHA and etag to AndroidApp and update package_name

### PR DESCRIPTION
**Pull request by mraouffouad**
1) Add SHA1, SHA256 hashes and etag fields to AndroidApp, 2) Update package_name suffix as it always needs to start with a letter.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
android_app: Added general fields `sha1_hashes`, `sha256_hashes` and `etag` to `google_firebase_android_app`.
```

```release-note:bug
android_app: Modify the `package_name` field suffix to always start with a letter in `google_firebase_android_app`.
```

### CODE REVIEW

File: mmv1/products/firebase/api.yaml [hunk 1]
```diff
@@ -204,25 +204,41 @@ objects:
...
+        name: sha1Hashes
+        description: |
+          The SHA1 certificate hashes for the AndroidApp.
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::Array
+        name: sha256Hashes
+        description: |
+          The SHA256 certificate hashes for the AndroidApp.
+        item_type: Api::Type::String  
+      - !ruby/object:Api::Type::String
```
Comment (ScottSuarez) [inline mmv1/products/firebase/api.yaml:236]: This should be of type fingerprint.... making this field output only doesn't make sense.. If we are to include this field we need to send this to the api.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/products/firebase/api.yaml (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/products/firebase/api.yaml:236]: referring to etag

Comment (mraouffouad) [reply mmv1/products/firebase/api.yaml]: Done.

**Comment by mraouffouad** [comment]
> Tests passed during RECORDING mode: `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7066/artifacts/fca736c7-54ef-44d6-8ead-916ef8cd2550/recording/TestAccFirebaserulesRelease_BasicRelease.log)] `TestAccContainerCluster_withInvalidGatewayApiConfigChannel`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7066/artifacts/fca736c7-54ef-44d6-8ead-916ef8cd2550/recording/TestAccContainerCluster_withInvalidGatewayApiConfigChannel.log)]
> 
> Tests failed during RECORDING mode: `TestAccFirebaseAndroidApp_update`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7066/artifacts/fca736c7-54ef-44d6-8ead-916ef8cd2550/build-log/recording_build/TestAccFirebaseAndroidApp_update_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7066/artifacts/fca736c7-54ef-44d6-8ead-916ef8cd2550/recording/TestAccFirebaseAndroidApp_update.log)]
> 
> Please fix these to complete your PR View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7066/artifacts/fca736c7-54ef-44d6-8ead-916ef8cd2550/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7066/artifacts/fca736c7-54ef-44d6-8ead-916ef8cd2550/recording) for each test

@ScottSuarez the failing test is getting RESOURCE_EXHAUSTED. Could you clear up some apps for that CI project? We had the same issue on the past and the reviewer resolved it by doing so. Thanks!

**Comment by mraouffouad** [comment]
> Hello! I am a robot who works on Magic Modules PRs.
> 
> I've detected that you're a community contributor. @ScottSuarez, a repository maintainer, has been assigned to assist you and help review your changes.
> 
> ❓ First time contributing? Click here for more details

**Comment by mraouffouad** [comment]
@ScottSuarez This PR is ready now. PTAL.

**Review by ScottSuarez — APPROVED** [review]

**Comment by ScottSuarez** [comment]
Could you please correct the release notes to build inline with the style guide?

https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md

**Comment by mraouffouad** [comment]
> Could you please correct the release notes to build inline with the style guide?
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md

@ScottSuarez Done. Thanks!

**Comment by mraouffouad** [comment]
@ScottSuarez Thanks for reviewing. This PR needs to be merged.

**Comment by ScottSuarez** [comment]
merged !

**Pull request merged**